### ISSUE TITLE
How can I best indicate how a Collection was filtered?

### ISSUE THREAD

**Issue by aschrijver** [w3c/activitystreams#483]
Please Indicate One: 
 * [ ] Editorial
 * [x] Question
 * [ ] Feedback
 * [ ] Blocking Issue
 * [ ] Non-Blocking Issue

Please Describe the Issue:

When a (Ordered)Collection is returned by an API call, it will oftentimes be the result of a query, and only contain a filtered subset of the full collection, e.g. 'give me all items filtered by Object type' or 'give me all items published between X and Y date', etc.

My question: If I want to convey to the client what filter query was used to obtain the collection, what would be the best practice?

Some options:

- The `id` property should be the indicator, so you can dereference the filtered collection reliably
  - But this is inconsistent with `inbox` and `outbox` in ActivityPub that _should_ be filtered by client permissions, and this is not reflected in the `id` (or anywhere else in the json).
- Another `object` property indicates the filter, like a special `tag`, or the `generator` property
- If the collection was represented as a Link, then the `rel` property could be an indicator of the filter
- There is no best way, choose whatever suits you

**Comment by nightpool** [comment]
i think `id` is probably your best bet, and I don't see any conflict with "should be filtered by client permissions"—that's just saying "don't show clients activities they don't have access to"

**Comment by aschrijver** [comment]
Thanks @nightpool 

I'll strongly consider `id` (maybe in combination with another option, if that works better for e.g. UI representation in a client)

**Comment by gobengo** [comment]
IMO "There is no best way, choose whatever suits you"

**Comment by evanp** [comment]
I just commented on the similar issue on sorting, so I'll link it here.

https://github.com/w3c/activitystreams/issues/484#issuecomment-1812959870

tl;dr is to use [FEP 5bf0](https://codeberg.org/fediverse/fep/src/branch/main/fep/5bf0/fep-5bf0.md), possibly with multiple types.

**Comment by aschrijver** [comment]
I am fine with closing this issue and its related one, with that reference to the FEP as the outcome.
